test/test_storage.py

```diff
@@ -312,17 +318,26 @@ def upload_progress(progress):
         st.delete_key(key)
 
 
+# sftp test support is available in vagrant, so can test just like for local if running in vagrant
 def _test_storage_init(storage_type, with_prefix, tmpdir, config_overrides=None):
     if storage_type == "local":
         storage_config = {"directory": str(tmpdir.join("rohmu"))}
+    elif storage_type == "sftp" and os.path.isfile('/etc/vagrantvm'):
+        # for no prefix testing, we need to cleanup existing files
+        os.system("sudo rm -rf /home/u12345678/*")
+
+        with open('/etc/sftp-password', 'r') as pwdfile:
```

Comment:
Nitpick: Looks like `"` should be preferred over `'`. Also, I'd reverse the order so that `rm -rf` is done only after successfully reading the password to make it even more unlikely this ends up blowing away some poor `u12345678` user's data on a machine that happened to have `/etc/vagrantvm` file.